htmx-spring-boot/src/main/java/io/github/wimdeblauwe/htmx/spring/boot/mvc/HtmxHandlerInterceptor.java

```diff
@@ -23,11 +24,16 @@ public boolean preHandle(HttpServletRequest request,
             Method method = ((HandlerMethod) handler).getMethod();
             setHxTrigger(response, method);
             setHxRefresh(response, method);
+            setVary(response, method);
         }
 
         return true;
     }
 
+    private void setVary(HttpServletResponse response, Method method) {
+        response.addHeader(HttpHeaders.VARY, HtmxRequestHeader.HX_REQUEST.getValue());
+    }
```

Comment:
Correct me if I'm wrong, but this adds `HX-Request` to the `VARY` header for all responses, whether it's necessary or not. Is that what we want?